OpenRobertaRobot/src/main/java/de/fhg/iais/roberta/visitor/validate/AbstractProgramValidatorVisitor.java

```diff
@@ -452,10 +452,7 @@ private void checkForZeroSpeedInCurve(Expr<Void> speedLeft, Expr<Void> speedRigh
         if ( speedLeft.getKind().hasName("NUM_CONST") && speedRight.getKind().hasName("NUM_CONST") ) {
             Double speedLeftNumConst = Double.valueOf(((NumConst<Void>) speedLeft).getValue());
             Double speedRightNumConst = Double.valueOf(((NumConst<Void>) speedRight).getValue());
-            int signLeft = (int) Math.signum(speedLeftNumConst);
-            int signRight = (int) Math.signum(speedRightNumConst);
-            boolean sameSpeed = Math.abs(speedLeftNumConst) == Math.abs(speedRightNumConst); //NOSONAR : TODO: supply an delta of 0.1 (speed is in [0,100] ?
-            if ( sameSpeed && (signLeft != signRight) && (signLeft != 0) && (signRight != 0) ) {
+            if ( (speedLeftNumConst == 0) && (speedRightNumConst == 0) ) {
```

Comment:
Here you are comparing a `double` to an `int` directly. In our case this is fine as the `double` is set to 0 directly from the block. However, this should be usually avoided.
Can you add a comparison with a delta instead (not directly `x == 0` because `x` may be 0.00000001)?